src/TextContent.ts

```diff
@@ -161,6 +161,16 @@ module Shumway {
           }
         },
         start: (tagName, attributes) => {
+          var style = null;
+          if (styleSheet) {
+            style = styleSheet.getStyle(tagName);
```

Comment:
`getStyle` is a hugely expensive operation, and we end up essentially reverting everything it does in `textFormat.merge` right below. How about we instead introduce an internal `StyleSheet#_getRawStyle` function that returns the object unmodified, and then have a `TextFormat#mergeStyleDefinition` method to use these? That'd just need iterate over the properties on the style definition and copy those over that are valid format properties.